### Cleaning a dataset
Clean the dataset replace missing values, remove outliers etc

Dataset :- https://docs.google.com/spreadsheets/d/e/2PACX-1vTSSTcErkXNk8KB0AlijhitwetxeHD2M3R0HJl2QPMAyFq0fxFX4PFKnzAWLDnratIz67DNL6GsZnV/pub?output=csv

In [1]:
# import the libraries
import pandas as pd

In [2]:
# load the dataset
dataset = pd.read_csv(r"C:\Users\Harbiodun\Downloads\dataset - netflix1 (1).csv")

In [3]:
# print out the dataset
dataset

show_id     type                             title         director  \
0         s1    Movie              Dick Johnson Is Dead  Kirsten Johnson   
1         s3  TV Show                         Ganglands  Julien Leclercq   
2         s6  TV Show                     Midnight Mass    Mike Flanagan   
3        s14    Movie  Confessions of an Invisible Girl    Bruno Garotti   
4         s8    Movie                           Sankofa     Haile Gerima   
...      ...      ...                               ...              ...   
8785   s8797  TV Show                        Yunus Emre        Not Given   
8786   s8798  TV Show                         Zak Storm        Not Given   
8787   s8801  TV Show                Zindagi Gulzar Hai        Not Given   
8788   s8784  TV Show                              Yoko        Not Given   
8789   s8786  TV Show                               YOM        Not Given   

            country  date_added  release_year rating   duration  \
0     United States   9/25/2021          2020  PG-13     90 min   
1            France   9/24/2021          2021  TV-MA   1 Season   
2     United States   9/24/2021          2021  TV-MA   1 Season   
3            Brazil   9/22/2021          2021  TV-PG     91 min   
4     United States   9/24/2021          1993  TV-MA    125 min   
...             ...         ...           ...    ...        ...   
8785         Turkey   1/17/2017          2016  TV-PG  2 Seasons   
8786  United States   9/13/2018          2016  TV-Y7  3 Seasons   
8787       Pakistan  12/15/2016          2012  TV-PG   1 Season   
8788       Pakistan   6/23/2018          2016   TV-Y   1 Season   
8789       Pakistan    6/7/2018          2016  TV-Y7   1 Season   

                                              listed_in  
0                                         Documentaries  
1     Crime TV Shows, International TV Shows, TV Act...  
2                    TV Dramas, TV Horror, TV Mysteries  
3                    Children & Family Movies, Comedies  
4      Dramas, Independent Movies, International Movies  
...                                                 ...  
8785                  International TV Shows, TV Dramas  
8786                                           Kids' TV  
8787  International TV Shows, Romantic TV Shows, TV ...  
8788                                           Kids' TV  
8789                                           Kids' TV  

[8790 rows x 10 columns]

In [4]:
# print out data type of each column
dataset.dtypes

show_id         object
type            object
title           object
director        object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
dtype: object

In [5]:
# print out sum of rows which are null
dataset.isna().sum()

show_id         0
type            0
title           0
director        0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
dtype: int64

In [6]:
# print out the columns in the dataset
dataset.columns

Index(['show_id', 'type', 'title', 'director', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in'],
      dtype='object')

In [7]:
# print out all the unique names of the director
for a in dataset['director'].unique():
    print(a)

Kirsten Johnson
Julien Leclercq
Mike Flanagan
Bruno Garotti
Haile Gerima
Andy Devonshire
Theodore Melfi
Suhas Kadav
Christian Schwochow
Not Given
Krysia Plonka, Kristian Mercado
Mark Thornton, Todd Kauffman
Robert Cullen, José Luis Ucha
Kongkiat Komesiri
Pedro de Echave García, Pablo Azorín Williams
Anirban Majumder
Olivier Megaton
Alex Woo, Stanley Moore
S. Shankar
Dennis Dugan
Scott Stewart
Robert Luketic
Rajiv Menon
Adam Salky
K.S. Ravikumar
Ashwiny Iyer Tiwari, Abhishek Chaubey, Saket Chaudhary
Daniel Sandu
Cédric Jimenez
George Nolfi
Bunmi Ajakaiye
Ben Simms
Steven Spielberg
Jeannot Szwarc
Joe Alves
Joseph Sargent
Daniel Espinosa
Antoine Fuqua
Toshiya Shinohara
Tyler Greco
Masahiko Murata
Hajime Kamegaki
Hirotsugu Kawasaki
Toshiyuki Tsuru
Tensai Okamura
Kemi Adetiba
Cedric Nicolas-Troyan
JJC Skillz, Funke Akindele
Thomas Sieben
Alice Waddington
Mona Achache, Patricia Tourancheau
Raja Gosnell
Fredrick Munk
Prakash Satam
Tomer Eshed
Delhiprasad Deenadayalan
Marcus Clarke
Alexis Alms

Ashwin Saravanan
Julia Knowles
Tarsem Singh
Kepa Sojo
Robert Moresco
Vikramaditya Motwane, Anurag Kashyap
Cho Li, Chen Hung-yi, Weica Wang, Liu Bang-yao, Lin Guan-fu, Shen Chi, YC Tom Lee
Manu Ashokan
Anita Udeep
Vlad Yudin
Alberto Arnaut Estrada
Aadish Keluskar
Joe Murray, Cosmo Segurson
Jesse Moss
Camille Shooshani
Shin Won-ho
Sujoy Ghosh
John Asher
Alex Burunova
Jesús Torres Torres
Vijay Kumar
Phanindra Narsetti
Gideon Raff
Jorge M. Fontana
Olivier Afonso
Karim Amer, Jehane Noujaim
Jay Chapman
Jay Karas, Demetri Martin
Dante Lam
Chi Keung Fung
Corey Yuen, Patrick Leung
To-hoi Kong
Yoo Byung-jae
Frank Ariza
Jerry Seinfeld
Aditya Sarpotdar
Nisheeta Keni
Daniel Gray Longino
Gerardo Olivares
Wi Ding Ho
Joe Lynch
Andreas Pichler
Daniel Arasanz
Spike Jonze
Marcelo Altmark, Mariano Mucci, Luis A. Scalella
Hernán Zin
Sittisiri Mongkolsiri
Robbie Grewal
Thiagarajan Kumararaja
Kate Horne
Roberto Girault Facha
Alex Díaz, Marcos Bucay
Svati Chakravarty Bhatkal
Kevin Bray
Jesse V. Johnson, Jesse

Tony Zierra
James Wong
David R. Ellis
Gianfranco Rosi
Sabal Singh Shekawat
Arthur Muhammad
Nirpal Bhogal
Andrew Renzi
Gerry Blaksley
Cheh Chang
Marc Abraham
Kang Cheng, Shan Hua
Asri Bendacha
Juan Manuel Cotelo
Mark Helenowski, Kevin Pang
Dean Wright
Mark Murphy
Richard Miron
Jon Carey, Adam Darke
Fred M. Wilcox
Maya Zinshtein
Lee Fulkerson
Seth Gordon
Trudie Styler
Marc Levin
Wade Allain-Marcus, Jesse Allain-Marcus
Jude Okwudiafor Johnson
Marcus Nispel
Simon Verhoeven
Heber Cannon
Alastair Fothergill
Bob Rose
Le Van Kiet
Adam Del Giudice
Hoyt Yeatman
Fellipe Barbosa
Pavel Parkhomenko
Ifa Isfansyah
Rudy Soedjarwo
Hiroshi Katagiri
Manbhavan Singh
P.K. Baaburaaj
David Grossman
Guy Vasilovich
Nicholas Stoller
David Silberg
Rich Ragsdale
Kazuchika Kise, Masahiko Murata
Kazuchika Kise
Bille Woodruff
Partho Ghosh
Vincente Minnelli, Charles Walters
Sally Potter
Brett Whitcomb
Harold Cronk
Jedd Wider, Todd Wider
Gordon Chan
Michael Mason
Francis Lee
Takuya Inaba
Sam de Jong
Ivan Sen
Mark Landr

In [8]:
# print out all the unique movie type
for a in dataset['type'].unique():
    print(a)

Movie
TV Show


In [9]:
# print out all the unique title names
for a in dataset['title'].unique():
    print(a)

Dick Johnson Is Dead
Ganglands
Midnight Mass
Confessions of an Invisible Girl
Sankofa
The Great British Baking Show
The Starling
Motu Patlu in the Game of Zones
Je Suis Karl
Motu Patlu in Wonderland
Motu Patlu: Deep Sea Adventure
Motu Patlu: Mission Moon
99 Songs (Tamil)
Bridgerton - The Afterparty
Bling Empire - The Afterparty
Cobra Kai - The Afterparty
The Circle - The Afterparty
Jailbirds New Orleans
Crime Stories: India Detectives
True: Winter Wishes
True: Magical Friends
True: Wonderful Wishes
Dance & Sing with True
My Little Pony: A New Generation
Bangkok Breaking
Europe's Most Dangerous Man: Otto Skorzeny in Spain
Samudri Lootere
Monsters Inside: The 24 Faces of Billy Milligan
Go! Go! Cory Carson: Chrissy Takes the Wheel
Jeans
Grown Ups
Dark Skies
Paranoia
Jaguar
Chicago Party Aunt
Squid Game
Tayo and Little Wizards
The World's Most Amazing Vacation Rentals
Metal Shop Masters
Pokémon Master Journeys: The Series
Titipo Titipo
Mighty Raju
Tobot Galaxy Detectives
Hotel Del Luna
Bra

Pet Sematary 2
Quarantine Tales
Fireplace 4K: Crackling Birchwood from Fireplace for Your Home
Red Joan
Seven Pounds
She's Out of My League
Snow Day
Star Trek
Stuart Little
Sword of Trust
Talladega Nights: The Ballad of Ricky Bobby
The American
The Best of Enemies
The Game
The Karate Kid
The Karate Kid Part II
The Karate Kid Part III
The Life of David Gale
The Long Riders
The Queen
The Strangers
The Sum of All Fears
The Tourist
Underworld
Underworld: Awakening
Underworld: Rise of the Lycans
Voiceless
What Dreams May Come
Why Do Fools Fall in Love
Working Title
America: The Motion Picture
Deranged
Lying and Stealing
Legend of Exorcism
Mommy Issues
Prime Time
Sophie: A Murder in West Cork
A Way Back Home
Wanted
Killing Them Softly
Into the Wind
Wonder Boy
Here Comes the Rain
The Ice Road
Ray
The Seventh Day
Osuofia in London II
O Vendedor de Sonhos
Osuofia in London
The House of Flowers: The Movie
#Selfie
#Selfie 69
Love Is a Story
Good on Paper
Murder by the Coast
Broken
Cleo & Cuquin
R

Lee Daniels' The Butler
Nocturnal Animals
The Ripper
Vir Das: Outside In - The Lockdown Special
A Trash Truck Christmas
Grizzlies
The Professor and the Madman
A California Christmas
Canvas
Giving Voice
The Prom
Torbaaz
Life's Speed Bump
Song Exploder
Futmalls.com
Funny Boy
Just The Way You Are
She's the One
The Panti Sisters
Kalel, 15
Chico Bon Bon and the Very Berry Holiday
Rose Island
André & his olive tree
Bobbleheads The Movie
Spirit Riding Free: Ride Along Adventure
Triple 9
Super Monsters: Santa’s Super Monster Helpers
Ava
Room 2806: The Accusation
Detention
Mighty Express: A Mighty Christmas
Captain Underpants Mega Blissmas
Christmas Crossfire
MANK
You Are My Home
Between Maybes
Break
Just Another Christmas
Must Be... Love
Ari Eldjárn: Pardon My Icelandic
Fierce
3 Days to Kill
Angela's Christmas Wish
Angels & Demons
Chef
Effie Gray
Monster House
Natalie Palamides: Nate - A One Man Show
Peppermint
Hazel Brugger: Tropical
Quigley Down Under
The Da Vinci Code
The Happytime Murders


Music Teacher
Someone Great
The Ruthless
A Little Chaos
Franco Escamilla: Bienvenido al mundo
About Time
HOMECOMING: A film by Beyoncé
Bollywood Calling
Ek Khiladi Ek Haseena
Generation Iron 3
Hazaaron Khwaishein Aisi
Jhankaar Beats
Mumbai Matinee
Rainbow Jelly
Shabd
Shortcut Safari
Thriller
Who Would You Take to a Deserted Island?
Dabbe 5: Zehr-i Cin
Dabbe: The Possession
The Perfect Date
Persona
Liss Pereira: Reteniendo líquidos
The Silence
Petta (Telugu Version)
Sarvam Thaala Mayam (Telugu Version)
Alien Warfare
Petta
Unicorn Store
Bert Kreischer: The Machine
New Girl
Ricardo Quevedo: Los amargados somos más
Suzzanna: Buried Alive
Ek Ladki Ko Dekha Toh Aisa Laga
Jagat
Kevin Hart: Irresponsible
Love O2O
Pinky Memsaab
The Trap
Mercy Black
The Burial of Kojo
15-Aug
Bayonet
The Highwaymen
The Legend of Cocaine Island
Danur: I Can See Ghosts
Verses of Love
Nate Bargatze: The Tennessee Kid
Sei
Superlopez
Oversize Cops
Mirage
ReMastered: The Miami Showband Massacre
The Dirt
The Amazing Pra

Feel Rich
Fiddler on the Roof
Fifty
Figaro Pho
Fishpeople
Filmworker
Filosofi Kopi The Movie
Final Destination
Final Destination 2
Final Destination 3
Fire at Sea
Fireflies
First Impression
First Kill
FirstBorn
Fishtail
Fittest in Dubai
Five Elements Ninjas
Flash of Genius
Flimflam
Flock of Four
LA 92
Flying Guillotine 2
Follow Me
Footprints: The Path of Your Life
For Colored Girls
For Grace
For Greater Glory
For Love or Money
For the Birds
For the Love of Benji
Forbidden Games: The Justin Fashanu Story
Forbidden Planet
Forever Chape
Forever Pure
Forks Over Knives
Four Christmases
Four Weddings and a Funeral
Fracture
Frances Ha
Freak Show
Fred Claus
Free Willy
Freeway: Crack in the System
Freezer's Campaign
French Dirty
Freshman Year
Friday the 13th
Friend Request
From Japan to Egypt
From Paris with Love
Froning: The Fittest Man in History
Frozen Planet
Friday Night Tykes
Frozen River
Full Out
Functional Fitness
Furie
Furthest Witness
G-Force
Gabriel and the Mountain
Gagarin: First in 

Love Daily
My Dead Ex
The Unsettling
Total Drama
Zac and Mia
We Are: The Brooklyn Saints
50M2
BONDING
Outlander
Go Dog Go
Blown Away
Busted!
Demon Slayer: Kimetsu no Yaiba
Octonauts
Daughter From Another Mother
Spycraft
Hello Ninja
Bling Empire
Carmen Sandiego
Disenchantment
The Magicians
Creating The Queen's Gambit
Inside the World’s Toughest Prisons
Mighty Little Bheem: Kite Festival
Pretend It’s a City
Stuck Apart
Surviving Death
History of Swear Words
Nailed It! Mexico
QB1: Beyond the Lights
Cobra Kai
Dream Home Makeover
Headspace Guide to Meditation
Jenni Rivera: Mariposa de Barrio
Oddbods
Monarca
The Haunted Hathaways
The Minimalists: Less Is Now
Chilling Adventures of Sabrina
Best Leftovers Ever!
Dare Me
Equinox
Still Game
A Love So Beautiful
Sakho & Mangane
Regiment Diaries
Bridgerton
Lovestruck in the City
Rhyme Time Town Singalongs
We Are the Champions
Shaun the Sheep
Timmy Time
Diamond City
Home for Christmas
Morphle
Run On
How To Ruin Christmas
Marvel Anime: Wolverine
Marve

In [10]:
# print out all the distinct names of the country
for a in dataset['country'].unique():
    print(a)

United States
France
Brazil
United Kingdom
India
Germany
Pakistan
Not Given
China
South Africa
Japan
Nigeria
Spain
Philippines
Australia
Argentina
Canada
Hong Kong
Italy
New Zealand
Egypt
Colombia
Mexico
Belgium
Switzerland
Taiwan
Bulgaria
Poland
South Korea
Saudi Arabia
Thailand
Indonesia
Kuwait
Malaysia
Vietnam
Lebanon
Romania
Syria
United Arab Emirates
Sweden
Mauritius
Austria
Turkey
Czech Republic
Cameroon
Netherlands
Ireland
Russia
Kenya
Chile
Uruguay
Bangladesh
Portugal
Hungary
Norway
Singapore
Iceland
Serbia
Namibia
Peru
Mozambique
Ghana
Zimbabwe
Israel
Finland
Denmark
Paraguay
Cambodia
Georgia
Soviet Union
Greece
West Germany
Iran
Venezuela
Slovenia
Guatemala
Jamaica
Somalia
Croatia
Jordan
Luxembourg
Senegal
Belarus
Puerto Rico
Cyprus
Ukraine


In [11]:
# print out all the unique date added
for a in dataset['date_added'].unique():
    print(a)

9/25/2021
9/24/2021
9/22/2021
5/1/2021
9/23/2021
5/21/2021
7/13/2021
6/12/2021
5/7/2021
11/26/2019
6/15/2018
5/18/2018
6/18/2019
9/21/2021
9/20/2021
9/19/2021
9/17/2021
9/14/2021
9/10/2021
9/9/2021
9/7/2021
9/2/2021
9/1/2021
8/28/2021
8/29/2021
9/16/2021
4/7/2021
8/27/2021
9/15/2021
7/7/2021
8/25/2021
8/20/2021
8/18/2021
8/4/2021
8/3/2021
8/1/2021
7/30/2021
7/29/2021
7/28/2021
7/23/2021
7/9/2021
7/22/2021
7/21/2021
9/8/2021
4/14/2021
7/20/2021
7/14/2021
7/8/2021
7/2/2021
7/1/2021
6/24/2021
6/23/2021
9/11/2021
9/3/2021
6/22/2021
6/20/2021
6/4/2021
6/19/2021
9/5/2021
6/17/2021
9/4/2021
6/16/2021
6/15/2021
6/14/2021
6/11/2021
6/8/2021
5/26/2021
5/24/2021
5/22/2021
4/30/2021
4/29/2021
4/28/2021
9/6/2021
4/20/2021
4/16/2021
4/15/2021
4/11/2021
4/1/2021
3/19/2021
3/10/2021
3/5/2021
2/26/2021
2/25/2021
2/18/2021
1/4/2021
1/1/2021
12/31/2020
12/30/2020
12/7/2020
12/4/2020
11/20/2020
11/3/2020
10/30/2020
10/3/2020
1/29/2020
10/1/2020
9/25/2020
8/31/2021
9/19/2020
9/7/2020
9/2/2020
8/15/2020
7/2

3/3/2017
7/19/2017
8/16/2017
11/3/2015
7/8/2019
4/23/2017
4/17/2016
3/22/2015
7/7/2016
5/31/2015
3/28/2018
1/13/2016
5/4/2017
9/17/2016
9/22/2016
12/2/2015
3/4/2017
2/8/2016
5/25/2019
2/1/2015
8/13/2018
4/16/2015
4/20/2017
6/30/2015
10/5/2017
1/28/2018
5/22/2016
9/11/2019
4/1/2014


In [12]:
# print out all the unique release year
for a in dataset['release_year'].unique():
    print(a)

2020
2021
1993
2019
2013
2014
2018
1998
2010
2017
1997
1996
1975
1978
1983
1987
2012
2001
2002
2003
2004
2011
2008
2009
2007
2005
2006
2016
1994
2015
1982
1989
1990
1991
1999
1986
1984
1980
1961
2000
1995
1985
1925
1992
1976
1959
1988
1981
1972
1964
1954
1979
1958
1956
1963
1970
1973
1960
1974
1966
1971
1962
1969
1977
1967
1968
1965
1945
1946
1942
1955
1944
1947
1943


In [13]:
# print out all the unique movie ratings
for a in dataset['rating'].unique():
    print(a)

PG-13
TV-MA
TV-PG
TV-14
TV-Y7
TV-Y
PG
TV-G
R
G
NC-17
NR
TV-Y7-FV
UR


In [14]:
# print out all the duration of the movies
for a in dataset['duration'].unique():
    print(a)

90 min
1 Season
91 min
125 min
9 Seasons
104 min
87 min
127 min
76 min
71 min
131 min
39 min
36 min
34 min
35 min
46 min
67 min
65 min
61 min
166 min
103 min
97 min
106 min
2 Seasons
4 Seasons
147 min
94 min
161 min
111 min
110 min
105 min
96 min
124 min
116 min
98 min
115 min
122 min
55 min
99 min
88 min
100 min
6 Seasons
3 Seasons
23 min
102 min
93 min
95 min
85 min
83 min
182 min
5 Seasons
92 min
84 min
48 min
145 min
119 min
80 min
128 min
143 min
273 min
117 min
114 min
118 min
108 min
37 min
121 min
63 min
142 min
113 min
154 min
120 min
82 min
109 min
101 min
86 min
77 min
229 min
15 min
89 min
156 min
112 min
107 min
129 min
44 min
68 min
135 min
136 min
165 min
150 min
133 min
47 min
70 min
56 min
58 min
78 min
140 min
64 min
59 min
60 min
69 min
19 min
22 min
148 min
189 min
54 min
141 min
130 min
81 min
7 Seasons
138 min
132 min
8 Seasons
123 min
66 min
62 min
74 min
38 min
13 min
139 min
126 min
155 min
159 min
12 min
137 min
49 min
72 min
204 min
25 min
212 min
73 min
149 

In [15]:
# print out all the unique genre where the movie is listed in
for a in dataset['listed_in'].unique():
    print(a)

Documentaries
Crime TV Shows, International TV Shows, TV Action & Adventure
TV Dramas, TV Horror, TV Mysteries
Children & Family Movies, Comedies
Dramas, Independent Movies, International Movies
British TV Shows, Reality TV
Comedies, Dramas
Children & Family Movies, Comedies, Music & Musicals
Dramas, International Movies
Children & Family Movies, Music & Musicals
Dramas, International Movies, Music & Musicals
Movies
Comedies
Docuseries, Reality TV
British TV Shows, Crime TV Shows, Docuseries
Children & Family Movies
Kids' TV
Documentaries, International Movies
Crime TV Shows, Docuseries, International TV Shows
Comedies, International Movies, Romantic Movies
Horror Movies, Sci-Fi & Fantasy
Thrillers
International TV Shows, Spanish-Language TV Shows, TV Action & Adventure
TV Comedies
International TV Shows, TV Dramas, TV Thrillers
Reality TV
Anime Series, Kids' TV
Kids' TV, Korean TV Shows
International TV Shows, Romantic TV Shows, TV Comedies
International TV Shows, TV Action & Adventur

In [16]:
# count of each director names
dataset['director'].value_counts()

Not Given                         2588
Rajiv Chilaka                       20
Alastair Fothergill                 18
Raúl Campos, Jan Suter              18
Suhas Kadav                         16
                                  ... 
Matt D'Avella                        1
Parthiban                            1
Scott McAboy                         1
Raymie Muzquiz, Stu Livingston       1
Mozez Singh                          1
Name: director, Length: 4528, dtype: int64

From the above result, we can see that the names of director whose name were not given are 2588.
Since they are not given, they have to be removed.

In [17]:
# drop rows in which the director name is not given
dataset = dataset.where(dataset['director'] !='Not Given').dropna()

In [18]:
dataset

show_id     type                              title  \
0         s1    Movie               Dick Johnson Is Dead   
1         s3  TV Show                          Ganglands   
2         s6  TV Show                      Midnight Mass   
3        s14    Movie   Confessions of an Invisible Girl   
4         s8    Movie                            Sankofa   
...      ...      ...                                ...   
6565   s3351    Movie         True: Grabbleapple Harvest   
6566   s3509    Movie             True: Tricky Treat Day   
6567   s3554  TV Show       True and the Rainbow Kingdom   
6568   s4241    Movie  Sam Kinison: The Scream Continues   
6569   s1315  TV Show                   Chappelle's Show   

                          director        country  date_added  release_year  \
0                  Kirsten Johnson  United States   9/25/2021        2020.0   
1                  Julien Leclercq         France   9/24/2021        2021.0   
2                    Mike Flanagan  United States   9/24/2021        2021.0   
3                    Bruno Garotti         Brazil   9/22/2021        2021.0   
4                     Haile Gerima  United States   9/24/2021        1993.0   
...                            ...            ...         ...           ...   
6565  Mark Thornton, Todd Kauffman         Canada   11/1/2019        2019.0   
6566  Mark Thornton, Todd Kauffman         Canada   9/20/2019        2019.0   
6567  Mark Thornton, Todd Kauffman  United States   8/30/2019        2019.0   
6568              Walter C. Miller  United States  12/31/2018        2016.0   
6569                   Stan Lathan  United States   2/12/2021        2005.0   

     rating   duration                                          listed_in  
0     PG-13     90 min                                      Documentaries  
1     TV-MA   1 Season  Crime TV Shows, International TV Shows, TV Act...  
2     TV-MA   1 Season                 TV Dramas, TV Horror, TV Mysteries  
3     TV-PG     91 min                 Children & Family Movies, Comedies  
4     TV-MA    125 min   Dramas, Independent Movies, International Movies  
...     ...        ...                                                ...  
6565   TV-Y     24 min                           Children & Family Movies  
6566   TV-Y     24 min                           Children & Family Movies  
6567   TV-Y  3 Seasons                                           Kids' TV  
6568  TV-MA     29 min                                      Documentaries  
6569  TV-MA  2 Seasons                                        TV Comedies  

[6202 rows x 10 columns]

In [19]:
# count of each director names
dataset['country'].value_counts()

United States     2401
India              975
United Kingdom     406
Not Given          287
Canada             189
                  ... 
Zimbabwe             1
Mozambique           1
Namibia              1
Cameroon             1
Croatia              1
Name: country, Length: 79, dtype: int64

From the above result, we can see that that 287 country names were not given.
Since they are not given, they also need to be removed.

In [20]:
# drop rows in which the country value is not given
dataset = dataset.where(dataset['country']!='Not Given').dropna()

In [21]:
dataset

show_id     type                              title  \
0         s1    Movie               Dick Johnson Is Dead   
1         s3  TV Show                          Ganglands   
2         s6  TV Show                      Midnight Mass   
3        s14    Movie   Confessions of an Invisible Girl   
4         s8    Movie                            Sankofa   
...      ...      ...                                ...   
6565   s3351    Movie         True: Grabbleapple Harvest   
6566   s3509    Movie             True: Tricky Treat Day   
6567   s3554  TV Show       True and the Rainbow Kingdom   
6568   s4241    Movie  Sam Kinison: The Scream Continues   
6569   s1315  TV Show                   Chappelle's Show   

                          director        country  date_added  release_year  \
0                  Kirsten Johnson  United States   9/25/2021        2020.0   
1                  Julien Leclercq         France   9/24/2021        2021.0   
2                    Mike Flanagan  United States   9/24/2021        2021.0   
3                    Bruno Garotti         Brazil   9/22/2021        2021.0   
4                     Haile Gerima  United States   9/24/2021        1993.0   
...                            ...            ...         ...           ...   
6565  Mark Thornton, Todd Kauffman         Canada   11/1/2019        2019.0   
6566  Mark Thornton, Todd Kauffman         Canada   9/20/2019        2019.0   
6567  Mark Thornton, Todd Kauffman  United States   8/30/2019        2019.0   
6568              Walter C. Miller  United States  12/31/2018        2016.0   
6569                   Stan Lathan  United States   2/12/2021        2005.0   

     rating   duration                                          listed_in  
0     PG-13     90 min                                      Documentaries  
1     TV-MA   1 Season  Crime TV Shows, International TV Shows, TV Act...  
2     TV-MA   1 Season                 TV Dramas, TV Horror, TV Mysteries  
3     TV-PG     91 min                 Children & Family Movies, Comedies  
4     TV-MA    125 min   Dramas, Independent Movies, International Movies  
...     ...        ...                                                ...  
6565   TV-Y     24 min                           Children & Family Movies  
6566   TV-Y     24 min                           Children & Family Movies  
6567   TV-Y  3 Seasons                                           Kids' TV  
6568  TV-MA     29 min                                      Documentaries  
6569  TV-MA  2 Seasons                                        TV Comedies  

[5915 rows x 10 columns]

In [22]:
# print out the data type of each column
dataset.dtypes

show_id          object
type             object
title            object
director         object
country          object
date_added       object
release_year    float64
rating           object
duration         object
listed_in        object
dtype: object

In [23]:
dataset['date_added'] = pd.to_datetime(dataset['date_added'])

In [24]:
# convert release_year column to string
dataset['release_year'] = dataset['release_year'].astype('int64')

In [25]:
# print out the data type of each column
dataset.dtypes

show_id                 object
type                    object
title                   object
director                object
country                 object
date_added      datetime64[ns]
release_year             int64
rating                  object
duration                object
listed_in               object
dtype: object

In [26]:
dataset

show_id     type                              title  \
0         s1    Movie               Dick Johnson Is Dead   
1         s3  TV Show                          Ganglands   
2         s6  TV Show                      Midnight Mass   
3        s14    Movie   Confessions of an Invisible Girl   
4         s8    Movie                            Sankofa   
...      ...      ...                                ...   
6565   s3351    Movie         True: Grabbleapple Harvest   
6566   s3509    Movie             True: Tricky Treat Day   
6567   s3554  TV Show       True and the Rainbow Kingdom   
6568   s4241    Movie  Sam Kinison: The Scream Continues   
6569   s1315  TV Show                   Chappelle's Show   

                          director        country date_added  release_year  \
0                  Kirsten Johnson  United States 2021-09-25          2020   
1                  Julien Leclercq         France 2021-09-24          2021   
2                    Mike Flanagan  United States 2021-09-24          2021   
3                    Bruno Garotti         Brazil 2021-09-22          2021   
4                     Haile Gerima  United States 2021-09-24          1993   
...                            ...            ...        ...           ...   
6565  Mark Thornton, Todd Kauffman         Canada 2019-11-01          2019   
6566  Mark Thornton, Todd Kauffman         Canada 2019-09-20          2019   
6567  Mark Thornton, Todd Kauffman  United States 2019-08-30          2019   
6568              Walter C. Miller  United States 2018-12-31          2016   
6569                   Stan Lathan  United States 2021-02-12          2005   

     rating   duration                                          listed_in  
0     PG-13     90 min                                      Documentaries  
1     TV-MA   1 Season  Crime TV Shows, International TV Shows, TV Act...  
2     TV-MA   1 Season                 TV Dramas, TV Horror, TV Mysteries  
3     TV-PG     91 min                 Children & Family Movies, Comedies  
4     TV-MA    125 min   Dramas, Independent Movies, International Movies  
...     ...        ...                                                ...  
6565   TV-Y     24 min                           Children & Family Movies  
6566   TV-Y     24 min                           Children & Family Movies  
6567   TV-Y  3 Seasons                                           Kids' TV  
6568  TV-MA     29 min                                      Documentaries  
6569  TV-MA  2 Seasons                                        TV Comedies  

[5915 rows x 10 columns]

In [27]:
# How to capitalise the first word of each element in a list
dataset.columns = [a.title() for a in dataset.columns]

In [28]:
dataset

Show_Id     Type                              Title  \
0         s1    Movie               Dick Johnson Is Dead   
1         s3  TV Show                          Ganglands   
2         s6  TV Show                      Midnight Mass   
3        s14    Movie   Confessions of an Invisible Girl   
4         s8    Movie                            Sankofa   
...      ...      ...                                ...   
6565   s3351    Movie         True: Grabbleapple Harvest   
6566   s3509    Movie             True: Tricky Treat Day   
6567   s3554  TV Show       True and the Rainbow Kingdom   
6568   s4241    Movie  Sam Kinison: The Scream Continues   
6569   s1315  TV Show                   Chappelle's Show   

                          Director        Country Date_Added  Release_Year  \
0                  Kirsten Johnson  United States 2021-09-25          2020   
1                  Julien Leclercq         France 2021-09-24          2021   
2                    Mike Flanagan  United States 2021-09-24          2021   
3                    Bruno Garotti         Brazil 2021-09-22          2021   
4                     Haile Gerima  United States 2021-09-24          1993   
...                            ...            ...        ...           ...   
6565  Mark Thornton, Todd Kauffman         Canada 2019-11-01          2019   
6566  Mark Thornton, Todd Kauffman         Canada 2019-09-20          2019   
6567  Mark Thornton, Todd Kauffman  United States 2019-08-30          2019   
6568              Walter C. Miller  United States 2018-12-31          2016   
6569                   Stan Lathan  United States 2021-02-12          2005   

     Rating   Duration                                          Listed_In  
0     PG-13     90 min                                      Documentaries  
1     TV-MA   1 Season  Crime TV Shows, International TV Shows, TV Act...  
2     TV-MA   1 Season                 TV Dramas, TV Horror, TV Mysteries  
3     TV-PG     91 min                 Children & Family Movies, Comedies  
4     TV-MA    125 min   Dramas, Independent Movies, International Movies  
...     ...        ...                                                ...  
6565   TV-Y     24 min                           Children & Family Movies  
6566   TV-Y     24 min                           Children & Family Movies  
6567   TV-Y  3 Seasons                                           Kids' TV  
6568  TV-MA     29 min                                      Documentaries  
6569  TV-MA  2 Seasons                                        TV Comedies  

[5915 rows x 10 columns]

### Conclusion
From careful observation of the dataset; 
1. 2588 rows from the director column whose name were not given were removed.
2. 287 rows from the country column whose name were not given were removed.
3. The date_added column data type was converted from string to datetime
4. The release_year column data type was converted from float to integer
5. The dataset column names were properly formatted